disnake/abc.py

```diff
@@ -781,20 +798,6 @@ def permissions_for(
                 base.handle_overwrite(allow=overwrite.allow, deny=overwrite.deny)
                 break
 
-        # if you can't send a message in a channel then you can't have certain
-        # permissions as well
-        if not base.send_messages:
-            base.send_tts_messages = False
-            base.send_voice_messages = False
-            base.mention_everyone = False
-            base.embed_links = False
-            base.attach_files = False
-
-        # if you can't view a channel then you have no permissions there
-        if not base.view_channel:
-            denied = Permissions.all_channel()
-            base.value &= ~denied.value
```

Comment:
Doesn't this still apply in all cases?

Reply:
Yes, but its continuing to be applied after the implicit permissions.